### PR TITLE
added attach/detach functions to configurations

### PR DESCRIPTION
**Pull request by eugenetaranov**
Hello,

this pr adds functions that allow configurations to be attached or detached, please kindly review it.

For 544

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:
https://github.com/openstack/trove/blob/b269b78495ee117e32cb36e2cec48361129610e3/trove/instance/models.py#L1284
https://github.com/openstack/trove/blob/b269b78495ee117e32cb36e2cec48361129610e3/trove/instance/models.py#L1350

This is a contribution from [Ribose Inc.](https://www.ribose.com)

### CODE REVIEW

File: openstack/db/v1/configurations/testing/requests_test.go [hunk 1]
```diff
@@ -235,3 +235,21 @@ func TestGetGlobalParam(t *testing.T) {
 
 	th.AssertDeepEquals(t, expected, param)
 }
+
+func TestAttachConfigGroup(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+	fixture.SetupHandler(t, resURL, "PUT", "", "", 202)
+
+	_, err := configurations.AttachConfigGroup(fake.ServiceClient(), configID, "zzz").Extract()
```
Comment (jtopjian) [inline openstack/db/v1/configurations/testing/requests_test.go]: It looks like the function parameters have gotten mixed up here.

In the function signature you have:

```go
AttachConfigGroup(client *gophercloud.ServiceClient, id string, configID string)
```

So I would think that `configID` and `zzz` would be swapped?

File: openstack/db/v1/configurations/requests.go [hunk 2]
```diff
@@ -165,3 +165,17 @@ func GetGlobalParam(client *gophercloud.ServiceClient, versionID, paramID string
 	_, r.Err = client.Get(getGlobalParamURL(client, versionID, paramID), &r.Body, nil)
 	return
 }
+
+// AttachConfigGroup will attach configuration group to the instance
+func AttachConfigGroup(client *gophercloud.ServiceClient, id string, configID string) (r commonResult) {
```
Comment (jtopjian) [inline openstack/db/v1/configurations/requests.go]: `id` seems ambiguous here. From what I can tell `id` is the instance ID? If so, maybe rename `id` to `instanceID`.

File: openstack/db/v1/configurations/requests.go [hunk 3]
```diff
@@ -165,3 +165,17 @@ func GetGlobalParam(client *gophercloud.ServiceClient, versionID, paramID string
...
+	b := map[string]interface{}{"instance": map[string]interface{}{"configuration": configID}}
+	_, r.Err = client.Put(resourceURL(client, id), &b, nil, &gophercloud.RequestOpts{OkCodes: []int{202}})
+	return
+}
+
+// DetachConfigGroup will dettach configuration group from the instance
+func DetachConfigGroup(client *gophercloud.ServiceClient, id string) (r commonResult) {
```
Comment (jtopjian) [inline openstack/db/v1/configurations/requests.go]: Same note about `id` here.

File: openstack/db/v1/configurations/requests.go (hunk 2 above)
Comment (eugenetaranov) [reply openstack/db/v1/configurations/requests.go]: makes sense, thank you

File: openstack/db/v1/configurations/requests.go (hunk 3 above)
Comment (eugenetaranov) [reply openstack/db/v1/configurations/requests.go]: fixed

File: openstack/db/v1/configurations/testing/requests_test.go (hunk 1 above)
Comment (eugenetaranov) [reply openstack/db/v1/configurations/testing/requests_test.go]: yep, looks like that's wrong .. should it invoke also instance creation and send real id there?

**Comment by eugenetaranov** [comment]
Thanks @jtopjian, renamed var names.
About acceptance tests - let me read the docs and try them :) hopefully will get it working.

**Comment by jtopjian** [comment]
@eugenetaranov Quick note: don't worry about an acceptance test for this. I've started reviewing and cleaning up the db acceptance tests and it would be a good amount of work to add one in right now.

**Comment by eugenetaranov** [comment]
Thanks @jtopjian. Meanwhile I fixed tests so please review.

File: openstack/db/v1/instances/testing/requests_test.go [hunk 4]
```diff
@@ -132,3 +133,31 @@ func TestResizeVolume(t *testing.T) {
...
+func TestAttachConfigGroup(t *testing.T) {
+	th.SetupHTTP()
+	defer th.TeardownHTTP()
+
+	req := `{
+    	"instance": {
+        	"configuration": "00000000-0000-0000-0000-000000000000"
+    	}
+	}`
+	fixture.SetupHandler(t, resURL, "PUT", req, "", 202)
```
Comment (jtopjian) [inline openstack/db/v1/instances/testing/requests_test.go]: It looks like you're creating a "raw" HTTP request here. What we want to do is create an actual Gophercloud request. See [here](https://github.com/gophercloud/gophercloud/blob/master/openstack/db/v1/flavors/testing/requests_test.go#L88-L107) for an example. The other unit tests are all created in the same way, too, so you can browse the various `testing/requests_test.go` files for more examples. Let me know if you need help, though.

**Comment by eugenetaranov** [comment]
thanks for feedback, @jtopjian. Updated tests, please review

**Comment by eugenetaranov** [comment]
@jtopjian I have added ConfigurationResult as you advised and renamed functions/some variables, please review. I appreciate your help on making this PR better 👍

**Comment by jtopjian** [comment]
@eugenetaranov This is good to go! Thank you for your work on this.

**Comment by ronaldtse** [comment]
Thank you @eugenetaranov and @jtopjian on behalf of [Ribose Inc.](https://github.com/riboseinc)!

**Pull request merged**